### ISSUE TITLE
expectRevert succeeds if any part of the revert message is included

### ISSUE THREAD

**Issue by shanefontaine** [OpenZeppelin/openzeppelin-test-helpers#95]
*This was recreated using @openzeppelin/test-helpers v0.5.3*

Using any part of the error message string in an `expectRevert` allows a test to succeed. Imagine the following code that works as expected:

**Manager.sol**

    function setManager(address _mgr) public {
        require(managers[msg.sender] == true, "Must be manager");
        ...
    }

**ManagerTest.js**

    it("Should revert for a non-manager", async () => {
        const revertMsg = 'Must be manager'
        await expectRevert(manager.setManager(accounts[1], { from: accounts[0] }), revertMsg)
      })

However, this test will also pass if `revertMsg = 'Must be`, `revertMsg = 'Must'`, or even `revertMsg = 'M'`.

I would assume that the revert message string should be identical to the `expectRevert` message.

**Comment by nventuro** [comment]
Thanks for bringing this up Shane!

This has come up a couple times (see https://github.com/OpenZeppelin/openzeppelin-test-helpers/issues/44), the reason why we used to be lax with what we accepted was due to a lack of a standard for how the revert reason is emitted: we'd need to extract it from the error message and then test strict equality, and such parsing can be fragile. We could probably pull it off though.

That said, I think the current behavior is reasonable enough: I've never seen source code where a revert message is a preffix of some other one, so by sticking to best practices and including the full revert reason string in the tests there should be no false positives.
 
@shanefontaine did this cause any issues in your testing setup?
